cg/services/illumina_services/cleaning_services/clean_runs_service.py

```diff
@@ -0,0 +1,207 @@
+"""Service to clean Illumina run directories."""
+
+import logging
+from pathlib import Path
+
+from housekeeper.store.models import File
+
+from cg.apps.housekeeper.hk import HousekeeperAPI
+from cg.constants import SequencingFileTag
+from cg.constants.time import TWENTY_ONE_DAYS
+from cg.exc import (
+    IlluminaCleanRunError,
+    HousekeeperBundleVersionMissingError,
+    HousekeeperFileMissingError,
+)
+from cg.models.run_devices.illumina_run_directory_data import IlluminaRunDirectoryData
+from cg.store.models import (
+    IlluminaSequencingRun,
+    IlluminaSampleSequencingMetrics,
+)
+from cg.store.store import Store
+from cg.utils.files import remove_directory_and_contents
+from cg.utils.time import is_directory_older_than_days_old
+
+LOG = logging.getLogger(__name__)
+
+
+class IlluminaCleanRunsService:
+    """
+    Handles the cleaning of Illumina runs in the sequencing-runs and demultiplexed-runs directories.
+    Requirements for cleaning:
+            Sequencing run is older than 21 days
+            Sequencing run is backed up
+            Sequencing run is in StatusDB
+            Sequencing run has sequencing metrics in StatusDB
+            Sequencing run has fastq files in Housekeeper or
+                Sequencing run has SPRING files in Housekeeper
+                and
+                Sequencing run has SPRING metadata in Housekeeper
+            Sequencing run has a sample sheet in Housekeeper
+    """
+
+    def __init__(
+        self,
+        status_db: Store,
+        housekeeper_api: HousekeeperAPI,
+        sequencing_run_path: Path,
+        dry_run: bool,
+    ):
+        self.status_db: Store = status_db
+        self.hk_api: HousekeeperAPI = housekeeper_api
+        self.sequencing_run_dir_data = IlluminaRunDirectoryData(
+            sequencing_run_path=sequencing_run_path
+        )
+        self.dry_run: bool = dry_run
+        LOG.info(f"Trying to delete {sequencing_run_path}")
+
+    def delete_run_directory(self) -> None:
+        """
+        Delete the sequencing runs and demultiplexed runs directory if it fulfills all requirements.
+        Raises:
+            IlluminaCleanRunError when any exception is caught
+        """
+        try:
+            self.set_sample_sheet_path_from_housekeeper()
+            if self.can_run_directory_be_deleted():
+                if self.dry_run:
+                    LOG.debug(
+                        f"Dry run: Would have removed: {self.sequencing_run_dir_data.get_sequencing_runs_dir()}"
+                    )
+                    return
+                remove_directory_and_contents(
+                    self.sequencing_run_dir_data.get_sequencing_runs_dir()
+                )
```

Comment:
This is a change in logic compared to https://github.com/Clinical-Genomics/cg/pull/3349/files#r2011986932